src/porepy/applications/convergence_analysis.py

```diff
@@ -0,0 +1,439 @@
+"""Module containing a class for performing spatio-temporal convergence analysis."""
+from __future__ import annotations
+
+import warnings
+from copy import deepcopy
+from typing import Literal, Optional, Union
+
+import numpy as np
+from scipy import stats
+
+import porepy as pp
+from porepy.utils.txt_io import TxtData, export_data_to_txt
+
+
+class ConvergenceAnalysis:
+    """Perform spatio-temporal convergence analysis on a given model.
+
+    The class :class:`~ConvergenceAnalysis` takes a PorePy model and its parameter
+    dictionary to run a batch of simulations with successively refined mesh sizes and
+    time steps and collect the results (i.e., the data classes containing the errors) in
+    a list using the :meth:`~run_analysis` method. Levels of refinement (in time and
+    space) are given at instantiation.
+
+    Useful methods of this class include
+
+        - :meth:`run_analysis`: Run a batch of simulations with successively refined
+          mesh size and/or time steps.
+        - :meth:`~export_errors_to_txt`: Exports errors into a ``txt`` file.
```

Comment:
I thought it's only needed when you want to shorten a path. @vlipovac, could you please clarify?

Reply:
Hyperlinks are created by `:meth:` followed by namespace between \` \`.
The ~ is for having `export_errors_to_txt` in the HTML instead of  `<namespace>.export_errors_to_txt`.

When linking to members of a class **inside** the docstring of other members, neither the whole namespace nor ~ is required since Sphinx can parse the namespace through the membership relation.

So it is not necessary in this case.

~ can be used when linking to members of other objects, and by omitting it one can facilitate the navigation.
If a reader sees  `<namespace>.export_errors_to_txt` instead of only `export_errors_to_txt` , the hyperlinked text in the HTML will contain the module path and the reader knows where the file is.
It can become quite unreadable for long namespaces. So in the other case its a design choice.

Reply:
I see. Thanks for the clarification @vlipovac. I will remove the `~` then.